### ISSUE TITLE
FATAL ERROR: Could not find my-default.cnf

### ISSUE THREAD

**Issue by jarib** [boxen/puppet-mysql#57]
Any idea what could cause this?

```
Notice: /Stage[main]/Mysql::Config/Exec[init-mysql-db]/returns: FATAL ERROR: Could not find my-default.cnf
Notice: /Stage[main]/Mysql::Config/Exec[init-mysql-db]/returns:
Notice: /Stage[main]/Mysql::Config/Exec[init-mysql-db]/returns: If you compiled from source, you need to run 'make install' to
Notice: /Stage[main]/Mysql::Config/Exec[init-mysql-db]/returns: copy the software into the correct location ready for operation.
Notice: /Stage[main]/Mysql::Config/Exec[init-mysql-db]/returns:
Notice: /Stage[main]/Mysql::Config/Exec[init-mysql-db]/returns: If you are using a binary release, you must either be at the top
Notice: /Stage[main]/Mysql::Config/Exec[init-mysql-db]/returns: level of the extracted archive, or pass the --basedir option
Notice: /Stage[main]/Mysql::Config/Exec[init-mysql-db]/returns: pointing to that location.
Error: /opt/boxen/homebrew/bin/mysql_install_db       --verbose       --basedir=/opt/boxen/homebrew       --datadir=/opt/boxen/data/mysql       --tmpdir=/tmp returned 1 instead of one of [0]
Error: /Stage[main]/Mysql::Config/Exec[init-mysql-db]/returns: change from notrun to 0 failed: /opt/boxen/homebrew/bin/mysql_install_db       --verbose       --basedir=/opt/boxen/homebrew       --datadir=/opt/boxen/data/mysql       --tmpdir=/tmp returned 1 instead of one of [0]
```

**Comment by jarib** [comment]
Looks like the file is here: 

`/opt/boxen/homebrew/opt/mysql/support-files/my-default.cnf`

Not sure if it would be more correct to have `--basedir=/opt/boxen/homebrew/opt/mysql/`. I'll try a symlink for now.

**Comment by jarib** [comment]
These are the paths that are searched by `mysql_install_db` before giving up (with `--basedir=/opt/boxen/homebrew`):

/opt/boxen/homebrew/./my-default.cnf
/opt/boxen/homebrew/./my-default.cnf.exe
/opt/boxen/homebrew/./release/my-default.cnf.exe
/opt/boxen/homebrew/./debug/my-default.cnf.exe
/opt/boxen/homebrew/./relwithdebinfo/my-default.cnf.exe
/opt/boxen/homebrew/share/my-default.cnf
/opt/boxen/homebrew/share/my-default.cnf.exe
/opt/boxen/homebrew/share/release/my-default.cnf.exe
/opt/boxen/homebrew/share/debug/my-default.cnf.exe
/opt/boxen/homebrew/share/relwithdebinfo/my-default.cnf.exe
/opt/boxen/homebrew/share/mysql/my-default.cnf
/opt/boxen/homebrew/share/mysql/my-default.cnf.exe
/opt/boxen/homebrew/share/mysql/release/my-default.cnf.exe
/opt/boxen/homebrew/share/mysql/debug/my-default.cnf.exe
/opt/boxen/homebrew/share/mysql/relwithdebinfo/my-default.cnf.exe
/opt/boxen/homebrew/support-files/my-default.cnf
/opt/boxen/homebrew/support-files/my-default.cnf.exe
/opt/boxen/homebrew/support-files/release/my-default.cnf.exe
/opt/boxen/homebrew/support-files/debug/my-default.cnf.exe
/opt/boxen/homebrew/support-files/relwithdebinfo/my-default.cnf.exe

**Comment by seanknox** [comment]
Confirmed. Going to change `--basedir` as you suggested. Thanks!

**Comment by mikemcquaid** [comment]
What version of the module are/were you using?

**Comment by seanknox** [comment]
2.0.1 has the error mentioned in the issue. 2.0.0 looks ok at quick
glance/run.

**Comment by mikemcquaid** [comment]
I'd be very surprised if that's the case; in my experience it was exactly the opposite (2.0.0 shipped really brokenly unfortunately).

**Comment by seanknox** [comment]
@jarib @mikemcquaid I didn't get to this yet, sorry. I'll try to take a look when I can, but I'd love a PR that:

1. includes a new class test for `manifests/config.pp`—ideally everything in there, but at the very least the  `'init-mysql-db'` section. We want to catch this from happening again.
2. Fix for proper `basedir`.

**Comment by mikemcquaid** [comment]
Reverted https://github.com/boxen/puppet-mysql/pull/66.

**Comment by pcasaretto** [comment]
I'm having the same issue here. Tried multiple versions.

**Comment by mikemcquaid** [comment]
@pcasaretto Is this failing on the latest version for you? Can you `gist` your failure output? Thanks!

**Comment by pcasaretto** [comment]
To fix it temporarily I symlinked `/opt/boxen/homebrew/opt/mysql/my-default.cnf` to `/opt/boxen/my-default.cnf`.
Now I can't reproduce the error even though I uninstalled mysql and deleted both `/opt/boxen/my-default.cnf` and `/opt/boxen/my.cnf`.

**Comment by mikemcquaid** [comment]
@pcasaretto Please upgrade to the newest version of this module and `brew reinstall mysql` and see if you can reproduce.

**Comment by mikemcquaid** [comment]
But I'm 99% sure this is fixed so closing for now unless you can reproduce.

**Comment by mcauser** [comment]
I also received the error on version 2.1.2
Fixed by adding symlink in `/opt/boxen/homebrew/`:
`my-default.cnf -> /opt/boxen/homebrew/opt/mysql/support-files/my-default.cnf`

It seems the config file is now nested in `/support-files`. The existing symlink was missing this additional path.

**Comment by evjan** [comment]
@mikemcquaid Getting this error on 2.1.2, just like @mcauser reports

**Comment by mikemcquaid** [comment]
@evjan You may need to `brew reinstall mysql`

**Comment by evjan** [comment]
@mikemcquaid It happened on a brand new machine that didn't have mysql installed. Does that mean running `brew reinstall mysql` wouldn't work?

**Comment by mikemcquaid** [comment]
@evjan Does it happen again when you rerun `boxen`?

**Comment by evjan** [comment]
@mikemcquaid Yup, every time on my machine and many others in our org :)

**Comment by mikemcquaid** [comment]
Hmm, weird. I wonder why I/we can't reproduce this. Reopening.

**Comment by evjan** [comment]
We've temporarily worked around it by adding the following nasty hack before the `github "mysql", "2.1.2"` in our Puppetfile:

```
unless(File.exist?("/opt/boxen/homebrew/my-default.cnf"))
  `ln -s  /opt/boxen/homebrew/Cellar/mysql/5.6.25/support-files/my-default.cnf /opt/boxen/homebrew/my-default.cnf`
end
```

**Comment by mikemcquaid** [comment]
@evjan If you (or anyone else here) can work on a PR to fix it properly we'd be grateful. As I can't reproduce this personally I won't be personally working on a fix, I'm afraid. Thanks!

**Comment by evjan** [comment]
@mikemcquaid I'm trying right now. I just noticed something weird. Even though our Puppetfile specifies 2.1.2, the change in 7f81a6 is not in `opt/boxen/repo/shared/mysql/data/Darwin.yaml`. It instead looks like this:

```
mysql::configdir:          "%{::boxen::config::configdir}/mysql"
mysql::globalconfigprefix: "%{::boxen::config::homebrewdir}"
mysql::datadir:            "%{::boxen::config::datadir}/mysql"
mysql::bindir:             "%{::boxen::config::homebrewdir}/bin"
mysql::executable:         "%{::boxen::config::homebrewdir}/bin/mysqld_safe"
mysql::client:             "%{::boxen::config::homebrewdir}/bin/mysql"
mysql::logdir:             "%{::boxen::config::logdir}/mysql"
```

Any idea why? What are we doing wrong?

**Comment by mikemcquaid** [comment]
@evjan That change was reverted as it wasn't correct; it broke for our MySQL configuration and we're not overriding those variables anywhere.

**Comment by coreyja** [comment]
@evjan Any update on this? Running into the same thing and wondering if you figured anything out in the meantime. Thank!

**Comment by evjan** [comment]
@coreyja Sorry, I failed at fixing this and subsequently gave up :(

**Comment by jacobbednarz** [comment]
This has landed in #82. Please have a go at updating and testing against that.
